src/app/pages/ConnectDevicePage/index.tsx

```diff
@@ -0,0 +1,119 @@
+    <Box align="center" direction="row" gap="small" justify="center">
+      {success ? (
+        <StatusGood color="successful-label" size="30px" />
+      ) : (
+        <StatusCritical color="status-error" size="30px" />
+      )}
+      <Text weight="bold" size="large" textAlign="center">
+        {label}
+      </Text>
+    </Box>
+  )
+}
+
+export function ConnectDevicePage() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const [connection, setConnection] = useState<ConnectionStatus>('disconnected')
+  const handleConnect = async () => {
+    setConnection('connecting')
+    try {
+      const device = await requestDevice()
+      if (device) {
+        setConnection(device ? 'connected' : 'disconnected')
+        dispatch(importAccountsActions.enumerateAccountsFromLedger())
+      }
+    } catch (error: any) {
+      if (error.message.match(/No device selected/)) {
+        setConnection('disconnected')
+      } else {
+        setConnection('error')
+      }
+    }
+  }
+
+  return (
+    <Box
+      style={{ minHeight: '100vh' }}
+      justify="center"
+      align="center"
+      pad="medium"
+      background="background-back"
+    >
+      <Box
+        elevation="small"
+        round="5px"
+        background="background-front"
+        pad="large"
+        style={{ position: 'relative' }}
+      >
+        <Box margin={{ bottom: 'medium' }} align="center">
+          <img src={logotype} alt="Oasis" width="75" height="75" />
+        </Box>
+        <Header textAlign="center">{t('ledger.extension.grantAccess', 'Grant access to your Ledger')}</Header>
+        <Box gap="medium">
+          <ol>
+            <li>
+              {t('ledger.instructionSteps.connectLedger', 'Connect your Ledger device to the computer')}
+            </li>
+            <li>{t('ledger.instructionSteps.closeLedgerLive', 'Close Ledger Live app on the computer')}</li>
+            <li>{t('ledger.instructionSteps.openOasisApp', 'Open the Oasis App on your Ledger device')}</li>
+            <li>
+              {t(
+                'ledger.extension.instructionStep',
+                'Once device is connected continue the operation in the wallet app',
+              )}
+            </li>
+          </ol>
+
+          {connection === 'connecting' && (
+            <Box
+              style={{ position: 'absolute', top: 0, right: 0, bottom: 0, left: 0 }}
+              background={{
+                color: 'background-front',
+                opacity: 'medium',
+              }}
+              align="center"
+              justify="center"
+            >
+              <Spinner size="medium" />
+            </Box>
+          )}
+          {connection === 'connected' && (
+            <ConnectionStatusIcon label={t('ledger.extension.succeed', 'Device connected')} />
+          )}
+          {connection === 'error' && (
+            <ConnectionStatusIcon success={false} label={t('ledger.extension.failed', 'Connection failed')} />
```

Comment:
Errors aren't as good as before

| Before - Firefox extension | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/3758846/197903568-15b46808-7990-4497-a2b7-f8537751a2d3.png) | ![image](https://user-images.githubusercontent.com/3758846/197903232-8118e8e7-a7de-41d0-ac68-f1ff843ff02e.png) |

| Before - Chrome with blocked webusb | After |
| --- | --- |
| ![Peek 2022-10-26 02-02](https://user-images.githubusercontent.com/3758846/197904184-0fc8ce6b-0ada-45af-bd52-0d91ad8c972a.gif) | ![Peek 2022-10-26 02-00](https://user-images.githubusercontent.com/3758846/197904191-9a9ec988-db46-4450-b99f-933f9b8a0779.gif) |

Reply:
| Updated error in Firefox (reproduced by temporarily installing ext in `about:debugging#/runtime/this-firefox`) | Chrome with blocked webusb is still stuck (reproduced with `Don't allow sites to connect to USB devices` in chrome://settings/content/usbDevices) |
| --- | --- |
| ![Screenshot from 2022-11-08 03-40-27](https://user-images.githubusercontent.com/3758846/200462480-5271f22b-ea8a-4503-8bcf-23583582a186.png)  | ![Peek 2022-11-08 03-47](https://user-images.githubusercontent.com/3758846/200462861-b112818a-8db9-44fd-9a3e-e9b5cadcbad3.gif)  |

Reply:
No specific error is thrown in this case just generic "No device selected". I will show ErrorFormatter in this case